File: ios/ObjcEmptyProject/ObjcEmptyProject/ViewController.m
//
//  ViewController.m
//  ObjcEmptyProject
//

#import "ViewController.h"
#import "LynxViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.

    [self setupUI];
}

- (void)setupUI {
    self.view.backgroundColor = [UIColor whiteColor];

    UILabel *label = [UILabel new];
    label.text = @"Hello Lynx!";
    label.textColor = [UIColor blackColor];
    label.font = [UIFont systemFontOfSize:30];

    [self.view addSubview:label];

    UIButton *button = [UIButton buttonWithType:UIButtonTypeSystem];
    [button setTitle:@"Go" forState:UIControlStateNormal];
    [button setTitleColor:[UIColor systemBlueColor] forState:UIControlStateNormal];
    [button setTitleColor:[UIColor blueColor] forState:UIControlStateHighlighted];
    [button addTarget:self action:@selector(buttonTapped:) forControlEvents:UIControlEventTouchUpInside];

    [self.view addSubview:button];

    label.translatesAutoresizingMaskIntoConstraints = NO;
    button.translatesAutoresizingMaskIntoConstraints = NO;

    [NSLayoutConstraint activateConstraints:@[
        [label.topAnchor constraintEqualToAnchor:self.view.safeAreaLayoutGuide.topAnchor],
        [label.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor],

        [button.topAnchor constraintEqualToAnchor:label.bottomAnchor constant:20],
        [button.centerXAnchor constraintEqualToAnchor:self.view.centerXAnchor],
    ]];
}

- (void)buttonTapped:(UIButton *)sender {
    // TODO: change url to lynx bundle url here
    NSString *urlString = @"";
    if (urlString.length > 0) {
        LynxViewController *vc = [[LynxViewController alloc] initWithUrl:urlString];
        [self.navigationController pushViewController:vc animated:YES];
    } else {
        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Warning" message:@"Please modify url" preferredStyle:UIAlertControllerStyleAlert];
        [self presentViewController:alertController animated:YES completion:nil];
        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(2 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
            [alertController dismissViewControllerAnimated:YES completion:nil];
        });
    }
}


@end

File: ios/SwiftEmptyProject/SwiftEmptyProject/ViewController.swift
//
//  ViewController.swift
//  SwiftEmptyProject
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
//        super.viewDidLoad()
        // Do any additional setup after loading the view.
        setupUI()
    }

    private func setupUI() {
            self.view.backgroundColor = .white

            let label = UILabel()
            label.text = "Hello Lynx!"
            label.textColor = .black
            label.font = UIFont.systemFont(ofSize: 30)

            self.view.addSubview(label)

            let button = UIButton()
            button.setTitle("Go", for: .normal)
            button.setTitleColor(.systemBlue, for: .normal)
            button.setTitleColor(.blue, for: .highlighted)
            button.addAction(UIAction{ [weak self] _ in
                // TODO: change url to lynx bundle url here
                let url = ""
                guard url.count > 0 else {
                    let alertController = UIAlertController(
                        title: "Warning", message: "Please modify url", preferredStyle: .alert)
                    self?.present(alertController, animated: true, completion: nil)
                    DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
                        alertController.dismiss(animated: true, completion: nil)
                    }
                    return
                }

                let vc = LynxViewController(url: url)
                self?.navigationController?.pushViewController(vc, animated: true)

            }, for: .touchUpInside)
            self.view.addSubview(button)

            label.translatesAutoresizingMaskIntoConstraints = false
            button.translatesAutoresizingMaskIntoConstraints = false

            NSLayoutConstraint.activate([
                label.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor),
                label.centerXAnchor.constraint(equalTo: self.view.centerXAnchor),

                button.topAnchor.constraint(equalTo: label.bottomAnchor, constant: 20),
                button.centerXAnchor.constraint(equalTo: self.view.centerXAnchor),

            ])
        }

}

